### ISSUE TITLE
Host ranges not working as expected

### ISSUE THREAD

**Issue by href** [relex/aini#1]
I stumbled upon your library when working on a little inventory CLI and I think I found an error.

To illustrate I wrote a minimal example:
```go
package main

import (
	"fmt"
	"strings"

	"github.com/relex/aini"
)

func main() {
	// Load from string example
	inventoryReader := strings.NewReader(`foo-[1:2]-[a:c]-bar`)
	inventory, _ := aini.Parse(inventoryReader)

	for host, _ := range inventory.Hosts {
		fmt.Println(host)
	}
}

```

The output of this when run through `go run main.go` is as follows:
```
foo-1-[a:c]-bar
foo-2-[a:c]-bar
```

I think this is incorrect and should result in this output instead:
```
foo-1-a-bar
foo-1-b-bar
foo-1-c-bar
foo-2-a-bar
foo-2-b-bar
foo-2-c-bar
```

**Comment by bemyak** [comment]
Hi! Thank you for reporting this.
Yes, I agree that the output __should__ look like this, but we are trying to maintain compatibility with Ansible, which allows to have only one range in host.

So, this is kind of expected behavior, unfortunately.

**Comment by href** [comment]
Does Ansible document this restriction anywhere? At work we use ranges like this, which is how I found out. Maybe we are doing something special, and I can’t currently verify that, but it would surprise me.

Also the following source file from Ansible seems to loop over range matches and it does even have an example in the comment:

https://github.com/ansible/ansible/blob/308723c3ca1122363e419070f1fa1d76ff5611a9/lib/ansible/parsing/utils/addresses.py#L178

I can check in more detail tomorrow if you’d like, but I thought maybe you researched this already and can point me at some resource that underscores your point.

In any case, thanks for your help and for open sourcing your library.

**Comment by href** [comment]
Digging a tad further I also found some test cases which seem to confirm my suspicions:
 
https://github.com/ansible/ansible/blob/308723c3ca1122363e419070f1fa1d76ff5611a9/test/units/plugins/inventory/test_inventory.py#L82

**Comment by bemyak** [comment]
Yes, sorry, I messed it up with the number of characters inside pattern, I remember having weird bugs there like `[ab:cd]` being expanded to `a`, `b` and `c` :)

I'll take a look at that, than you for digging that out.

**Comment by bemyak** [comment]
Okay, should be fixed now in v1.1.4

**Comment by href** [comment]
Nice, thank you for fixing it so quickly 🥇
